doc/nrf/ug_memfault.rst

```diff
@@ -0,0 +1,207 @@
+  There are 15 prepopulated metrics shown for your devices.
+  You can access device level data to resolve bugs faster.
+
+See the `Memfault documentation <Memfault introduction_>`_ for more details.
+
+.. note::
+   Memfault has been integrated in |NCS| since the v1.6.0 release.
+
+Prerequisites
+*************
+
+Before you start the |NCS| integration with Memfault, make sure that the following prerequisites are completed:
+
+   * :ref:`Setup of nRF Connect SDK <getting_started>` environment.
+   * :ref:`Setup of nRF9160 DK <ug_nrf9160_gs>`.
+   * Sign up in the `Memfault registration page`_ and `create a new project in Memfault`_.
+     You will be directed to the Integration guide for Memfault.
+     You receive the ``project key`` here to set in the :kconfig:option:`CONFIG_MEMFAULT_NCS_PROJECT_KEY` Kconfig option.
+     After registration, you can connect up to 100 devices for free.
+
+Solution architecture
+*********************
+
+The SDK provides functionality to collect debug information from your devices in the form of coredumps, error traces, metrics, logs, and more.
+Memfault saves the data to a non-volatile storage, after which it is broken into chunks and forwarded to Memfault's cloud solution for further analysis.
+Communication with the Memfault Cloud is handled by APIs available in `Memfault-SDK`_, while the integration code of the SDK in |NCS| provides some additional functionality.
+The Memfault cloud reassembles the data before analyzing and de-duplicates the data.
+The overall fleet health of the issues can be examined through the dashboard of the Memfault web application.
+See the `Memfault MCU Guide`_ for more information.
+
+.. figure:: /images/Memfault_architecture.svg
+   :alt: Memfault Solution architecture
+
+   Memfault Solution architecture
+
+.. _using_memfault:
+
+Integration overview
+********************
+
+The SDK version is included in the `west manifest file`_ in |NCS| and is automatically downloaded when running ``west update``.
+By default, it is downloaded to ``nRF/modules/memfault-firmware-sdk/``.
+
+Complete the following steps to integrate Memfault into |NCS|:
+
+1. :ref:`Configure Memfault <ug_memfault_config>`
+#. :ref:`Track errors <ug_memfault_error>`
+#. :ref:`Collect metrices <ug_memfault_metrics>`
+
+.. _ug_memfault_config:
+
+Configuring Memfault
+====================
+
+Complete the following steps to configure Memfault:
+
+1. Add the following Kconfig option in your :file:`prj.conf` file:
+
+   .. code-block:: console
+
+      CONFIG_MEMFAULT=y
+      CONFIG_MEMFAULT_NCS_PROJECT_KEY=""
+
+   The APIs in the Memfault SDK can then be linked into your application.
+   You can get your project key after signing up with Memfault.
+   The :kconfig:option:`CONFIG_MEMFAULT_NCS_PROJECT_KEY` Kconfig option is located in the :file:`../overlay-memfault.conf` file.
```

Comment:
Cannot find any other reference to `overlay-memfault.conf` other than in the ATv2 app. If this is the case this line needs to be removed if the docs are intended to be generic.